### ISSUE TITLE
PeerSet::Collation is behaving weirdly, leading to slow block production

### ISSUE THREAD

**Issue by crystalin** [paritytech/polkadot#3565]
This is based on Polkadot v0.9.8 and didn't seem to appear in previous versions.
This happens on multiple relay/parachain networks (of different size/config/runtimes) that we internally host (but not on Moonriver that is hosted on Kusama), I'll pick one of them as an exemple.

Setup:
- 4 validators relay (4 validator assigned to the parachain group), using westend-local
- 2 collators using moonbase-local
(This network have been running for 10 months without having this issue before)

Observation:
- After few hours usually (no pattern observed so far), collators will report `DEBUG tokio-runtime-worker parachain::gossip-support: error=NotAValidator` 
- Parachain blocks are not included in the relay slowing down the overall block production (leading to a stall)

Temporary solution:
- Restart a validator will allow to get the blocks included again

How to reproduce:
- It is a bit hard to reproduce without having a full network, but I'm still working on testing polkadot-launch to make it happen there. I'll update this ticket once I have more information

Additional observations:
- Setting the validation_per_core to 3 (to make sure there are some validator rotation) made it worse. So rotating validator (expecting to re-start the connection with the collators) is not helping.
- The number of validators seem to impact the frequency this issue happen (our bigger networks seemed to be less affected)
- The PeerSet::Collation established connection on Prometheus seem to suggest that collators are only connecting to 1 validator of the group, and when the issue happens it seems they are not connected anymore (still to be verified)

**Comment by bkchr** [comment]
CC @eskimor

**Comment by eskimor** [comment]
Hi @crystalin !

The debug message seems unrelated and is expected and seems unrelated - a collator is not a validator after all. It should appear in the logs sooner than just after a few hours of operation though.

If you could verify whether it really is the collator/validator connection that causes the problem, that would help narrowing down the issue. A setup for reproducing the issue locally would of course be perfect.

**Comment by ordian** [comment]
It would be nice to verify the problem reproduces with master, not just 0.9.8.

**Comment by ordian** [comment]
This message should be reported on collators with `parachain=debug` and/or `parachain::gossip-support=debug` logs enabled when they have synced to the tip of the relay chain block and is expected.

**Comment by crystalin** [comment]
I've been running polkadot-launch for some hours already without being able to reproduce it. I believe there must be some network interactions for the bug to be triggered.
Is there any log we should enable to gather more logs on our other networks ? Some of them are experiencing constantly.

**Comment by eskimor** [comment]
Yeah, if you are sure the problem lies within the collator-validator communication you can enable debug logging for the parachain::collator-protocol target. ( RUST_LOG=parachain::collator-protocol=debug) on both the validators and the collators. That could help.
